tests/test_areadetector.py

```diff
@@ -116,6 +117,64 @@ class MyDetector(SimDetector):
     # TODO subclassing issue
 
 
+def test_invalid_plugins():
+    class MyDetector(SingleTrigger, SimDetector):
+        tiff1 = Cpt(TIFFPlugin, 'TIFF1:')
+        stats1 = Cpt(StatsPlugin, 'Stats1:')
+
+    det = MyDetector(prefix)
+    det.wait_for_connection()
+    det.tiff1.nd_array_port.put(det.cam.port_name.get())
+    det.stats1.nd_array_port.put('AARDVARK')
+
+    with pytest.raises(RuntimeError):
+        det.stage()
+
+    assert not det.validate_asyn_ports()
+    assert ['AARDVARK'] == det.missing_plugins()
+
+
+def test_get_plugin_by_asyn_port():
+    class MyDetector(SingleTrigger, SimDetector):
+        tiff1 = Cpt(TIFFPlugin, 'TIFF1:')
+        stats1 = Cpt(StatsPlugin, 'Stats1:')
+        roi1 = Cpt(ROIPlugin, 'ROI1:')
+
+    det = MyDetector(prefix)
+
+    det.tiff1.nd_array_port.put(det.cam.port_name.get())
+    det.roi1.nd_array_port.put(det.cam.port_name.get())
+    det.stats1.nd_array_port.put(det.roi1.port_name.get())
+
+    assert det.validate_asyn_ports()
+
+    assert det.tiff1 is det.get_plugin_by_asyn_port(det.tiff1.port_name.get())
+    assert det.cam is det.get_plugin_by_asyn_port(det.cam.port_name.get())
+    assert det.roi1 is det.get_plugin_by_asyn_port(det.roi1.port_name.get())
+
+
+def test_read_configuration_smoke():
+    class MyDetector(SingleTrigger, SimDetector):
+        stats1 = Cpt(StatsPlugin, 'Stats1:')
+        porc1 = Cpt(ProcessPlugin, 'ROI1:')
```

Comment:
> (the ones you would consider to be enums)

I found them by looking for descriptions that and an `enum_str` field in them.  Is calling them enums not fair?